.github/workflows/perform_build.yml

```diff
@@ -0,0 +1,75 @@
+# SPDX-License-Identifier: BSD-2-Clause
+# X-SPDX-Copyright-Text: (c) Copyright 2023 Advanced Micro Devices, Inc.
+
+name: "perform_build"
+
+on: push
+
+jobs:
+  extract_branch_info:
+    runs-on: ubuntu-latest
+    timeout-minutes: 1
+    outputs:
+      tcpdirectBranch: ${{ steps.tcpdirectBranch.outputs.branch }}
+      onloadBranch: ${{ steps.onloadBranch.outputs.branch }}
+    steps:
+      - name: tcpdirect
+        uses: actions/checkout@v4
+        with:
+          path: tcpdirect
+
+      - name: Extract TCPDirect branch name
+        id: tcpdirectBranch
+        run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> "$GITHUB_OUTPUT"
+
+      - name: Extract Onload branch name
+        id: onloadBranch
+        run: |
+          export tcpdirect_branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}
+          echo "branch=$(sed s/tcpdirect/onload/g <<< $tcpdirect_branch)" >> "$GITHUB_OUTPUT"
```

Comment:
Might we want to build against the `last_known_good` branches that are known to at least build onload? That being said, they may be less useful if we're able to maintain branches in a good state by using actions to ensure things build and pass tests before merge.